packages/core/parcel-bundler/src/assets/CSSAsset.js

```diff
@@ -172,13 +250,24 @@ class CSSAst {
     this.dirty = false;
   }
 
-  render() {
+  render(name) {
     if (this.dirty) {
-      this.css = '';
-      postcss.stringify(this.root, c => (this.css += c));
+      let {css, map} = this.root.toResult({
+        to: name,
+        map: {inline: false, annotation: false, sourcesContent: true}
+      });
+
+      this.css = css;
+
+      return {
```

Comment:
Perhaps we should refactor this to:
```Javascript
return [
  {
    type: 'css',
    sourceMap: map
  }
];
```

and create (or extend existing by looking at childBundles) the actual sourcemap bundles here: https://github.com/parcel-bundler/parcel/blob/master/packages/core/parcel-bundler/src/Bundle.js#L39

Probably a good idea to get some input from Devon as well on this as Pipeline is his design

Reply:
JSAsset does something like that already:
```js
return {
    js: code,
    map: this.sourceMap
};
```
But I struggle to find where is this actually used (and how to make it work for css)

Reply:
That's the same as:

```
return [
  {
    type: 'map',
    value: this.sourceMap
  },
  {
    type: 'js',
    value: code
  }
];
```

This was just for backwards compatibility

Reply:
@devongovett ?

Reply:
At first, I thought `SourceMapPackager.shouldAddAsset` could be used for this, but it doesn't get called at all.

Reply:
Yeah I think we will need to add a `map` property to renditions themselves rather than having a the map be a separate rendition since only one rendition of the same type is supported (and there would be no way to associate them). Then, we'll need to special case source maps in the Bundler to create the `map` bundles. Perhaps in `Bundle#addAsset` we could add the map to a sibling `map` bundle? https://github.com/parcel-bundler/parcel/blob/master/packages/core/parcel-bundler/src/Bundle.js#L39

Reply:
In `Bundle#addAsset`, the map isn't really available anymore (on `asset.generated`), because it got "filtered" here: https://github.com/parcel-bundler/parcel/blob/d7dc51f9eacd80aadd1e41888771033e447758f1/packages/core/parcel-bundler/src/Pipeline.js#L25-L27

Reply:
Oh right... hmm

Reply:
>  For now, maybe you could move this conversion from array to object into the Bundler after adding assets to the bundles (createBundleTree). We just need to make sure it is converted before running Packagers for backwards compatibility

1. Assets are added by createBundleTree via Bundle#addAsset
2. Transform asset.generated to generatedMap
3. Run packagers: JSAsset calls Bundle#addAsset for bundle loaders 🙈

Reply:
I've chosen a different approach (loading sourcemaps from previous asset stages doesn't work yet)

Reply:
How can/why should a bundle be a child bundle and a sibling bundle at the same time?

Reply:
@devongovett ?

Reply:
> I am not against adding hacks, but I'm also fine with waiting for this feature until Parcel 2 and perhaps starting on the implementation there instead. What do you think?

I've solved the previous issue, regarding something more cosmetic:
The first two files go together and the next two have differing names (but would work fine):
```
test.9ac3ef99.css
test.9ac3ef99.css.map
test.f2b3894d.js
test.23936829.js.map
```

A possible fix?
```diff
diff --git a/packages/core/parcel-bundler/src/Bundle.js b/packages/core/parcel-bundler/src/Bundle.js
index 16d44aee..94e08697 100644
--- a/packages/core/parcel-bundler/src/Bundle.js
+++ b/packages/core/parcel-bundler/src/Bundle.js
@@ -120,6 +120,8 @@ class Bundle {
   }

   getHashedBundleName(contentHash) {
+    if (this.type == 'map')
+      return this.parentBundle.getHashedBundleName(contentHash) + '.map';
     // If content hashing is enabled, generate a hash from all assets in the bundle.
     // Otherwise, use a hash of the filename so it remains consistent across builds.
     let basename = Path.basename(this.name);
```